zerver/models/realms.py

```diff
@@ -1039,7 +1056,9 @@ def realm_pre_and_post_delete_handler(*, instance: Realm, **kwargs: object) -> N
 
 
 def get_realm(string_id: str) -> Realm:
-    return Realm.objects.get(string_id=string_id)
+    return Realm.objects.select_related(
+        "can_create_public_channel_group", "can_create_public_channel_group__named_user_group"
+    ).get(string_id=string_id)
```

Comment:
Probably better to format this with one per line by putting a comma at the end of the line.